From c0ee3676db918a00490ab04a32f5f81106fff82a Mon Sep 17 00:00:00 2001
From: Xiaolei Wang <xiaolei.wang@windriver.com>
Date: Mon, 27 Jul 2020 16:22:09 +0800
Subject: [PATCH 2/2] arm64: dts: imx: Delete the flexspi driver dtb node and
 use the m25p80 driver dtb node

The flash driver of flexspi does not support the 4-byte address mode,
delete the flexspi driver dtb node and use the m25p80 driver dtb
node to support imx8mm borad. commit f5fc8e4f9a1a ("MLK-21960-3: arm64:
dts: enable fspi in imx8mm dts") from
https://source.codeaurora.org/external/imx/linux-imx.git

Signed-off-by: Xiaolei Wang <xiaolei.wang@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8mm-evk.dts     | 25 ++++++++++---------
 arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi | 12 ++++-----
 2 files changed, 18 insertions(+), 19 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts
index 2e51dceedec8..e292cf504682 100755
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts
@@ -520,18 +520,19 @@
 };
 
 &flexspi {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_flexspi0>;
-	status = "okay";
-
-	flash0: mt25qu256aba@0 {
-		reg = <0>;
-		#address-cells = <1>;
-		#size-cells = <1>;
-		compatible = "micron,mt25qu256aba";
-		spi-max-frequency = <80000000>;
-		spi-nor,ddr-quad-read-dummy = <6>;
-	};
+        pinctrl-names = "default";
+        pinctrl-0 = <&pinctrl_flexspi0>;
+        status = "okay";
+
+        flash0: mt25qu256aba@0 {
+                reg = <0>;
+                #address-cells = <1>;
+                #size-cells = <1>;
+                compatible = "jedec,spi-nor";
+                spi-max-frequency = <80000000>;
+                spi-tx-bus-width = <4>;
+                spi-rx-bus-width = <4>;
+        };
 };
 
 &i2c1 {
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi
index 012903f21302..af3f5ff2160d 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi
@@ -961,15 +961,13 @@
 	flexspi: spi@30bb0000 {
 		#address-cells = <1>;
 		#size-cells = <0>;
-		compatible = "fsl,imx8mm-flexspi";
+		compatible = "nxp,imx8mm-fspi";
 		reg = <0 0x30bb0000 0 0x10000>, <0 0x08000000 0 0x10000000>;
-		reg-names = "FlexSPI", "FlexSPI-memory";
+		reg-names = "fspi_base", "fspi_mmap";
 		interrupts = <GIC_SPI 107 IRQ_TYPE_LEVEL_HIGH>;
-		clocks = <&clk IMX8MM_CLK_QSPI_ROOT>;
-		clock-names = "fspi";
-		assigned-clock-rates = <80000000>;
-		assigned-clocks = <&clk IMX8MM_CLK_QSPI>;
-		assigned-clock-parents = <&clk IMX8MM_SYS_PLL1_400M>;
+		clocks = <&clk IMX8MM_CLK_QSPI_ROOT>,
+			 <&clk IMX8MM_CLK_QSPI_ROOT>;
+		clock-names = "fspi", "fspi_en";
 		status = "disabled";
 	};
 
-- 
2.17.1

